src/registrar/models/domain.py

```diff
@@ -219,17 +226,107 @@ def nameservers(self) -> list[tuple[str]]:
         Subordinate hosts (something.your-domain.gov) MUST have IP addresses,
         while non-subordinate hosts MUST NOT.
         """
-        # TODO: call EPP to get this info instead of returning fake data.
-        return [
-            ("ns1.example.com",),
-            ("ns2.example.com",),
-            ("ns3.example.com",),
-        ]
+        try:
+            hosts = self._get_property("hosts")
+        except Exception as err:
```

Comment:
In future after we implement this, we will likely want to change this so that it catches a specific registry error rather than a broad stroke one (as we'd want to log that seperately)

Reply:
It will throw a value error here, or a registry error so there will be a bit of error handling needed for those cases. But agreed that is  covered in the nameservers ticket and 433 which handles error handling